### PR TITLE
minor, non functional changes following dart best practices

### PR DESCRIPTION
**Pull request by iapicca**
I've applied minor changes following [dart best practices](https://dart.dev/guides/language/effective-dart)
- added assertions in the constructors
- made constant ant whatever could be constant
- cleaned up formatting, arrow functions, trailing commas, `var`, `final` in initialization
- fixed missing `dispose` in controllers

no breaking change has been applied

### CODE REVIEW

**Comment by MarcusTomlinson** [comment]
Hmm, why do these not turn up during pedantic analysis?

**Comment by iapicca** [comment]
@MarcusTomlinson 
I wish there was, but unfortunately not, I've just did manually.
do I need a sort of invitation to sign the [cla](https://ubuntu.com/legal/contributors/agreement)?

**Comment by MarcusTomlinson** [comment]
> @MarcusTomlinson
> I wish there was, but unfortunately not, I've just did manually.
> do I need a sort of invitation to sign the [cla](https://ubuntu.com/legal/contributors/agreement)?

I don’t believe you require an invitation. Just fill in the form I think.

File: ubuntu_desktop_installer/pubspec.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies:
     sdk: flutter
   flutter_localizations:
     sdk: flutter
-  intl:
+  intl: ^0.17.0
```
Comment (MarcusTomlinson) [inline ubuntu_desktop_installer/pubspec.yaml:28]: Should be: intl: ^0.17.0-nullsafety.2

**Review by MarcusTomlinson — CHANGES_REQUESTED** [review]

File: ubuntu_desktop_installer/pubspec.yaml (hunk 1 above)
Comment (iapicca) [reply ubuntu_desktop_installer/pubspec.yaml]: my bad, I thought it was obsolete due to newer version, I was wrong

**Comment by kevlar700** [comment]
Some of these wouldn't be picked up by pedantic analysis, because e.g. short arrow is controversial. Personally I think always using brackets with return is more consistent and sometimes required.

**Review by MarcusTomlinson — CHANGES_REQUESTED** [review]
CI is unhappy: https://github.com/canonical/ubuntu-desktop-installer/pull/18/checks?check_run_id=1833358190

**Comment by iapicca** [comment]
> CI is unhappy: https://github.com/canonical/ubuntu-desktop-installer/pull/18/checks?check_run_id=1833358190

sorry about that

**Comment by MarcusTomlinson** [comment]
@iapicca, so it appears there is a set of linter rules for Effective Dart :)

If we are to make the switch from pedantic to this (which I actually think we should), we best go all in.

Would you be up to fixing the rest of these analyser errors in your branch? https://github.com/canonical/ubuntu-desktop-installer/pull/18/checks?check_run_id=1834232513

**Comment by iapicca** [comment]
> @iapicca, so it appears there is a set of linter rules for Effective Dart :)
> 
> If we are to make the switch from pedantic to this (which I actually think we should), we best go all in.
> 
> Would you be up to fixing the rest of these analyser errors in your branch? https://github.com/canonical/ubuntu-desktop-installer/pull/18/checks?check_run_id=1834232513
> 
> Or would you prefer I address them?

I've done these 

<details>
<summary>lint for ubuntu_desktop_installer</summary>

```bash
   info • Prefer relative imports for files in `lib/` • ubuntu_desktop_installer/lib/keyboardlayoutpage.dart:6:8 • prefer_relative_imports X
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/keyboardlayoutpage.dart:94:38 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/keyboardlayoutpage.dart:94:60 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/keyboardlayoutpage.dart:134:47 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/keyboardlayoutpage.dart:134:69 • avoid_types_on_closure_parameters
   info • Sort directive sections alphabetically • ubuntu_desktop_installer/lib/main.dart:8:1 • directives_ordering
   info • Sort directive sections alphabetically • ubuntu_desktop_installer/lib/main.dart:10:1 • directives_ordering
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/main.dart:48:29 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/main.dart:50:27 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/main.dart:51:31 • avoid_types_on_closure_parameters
   info • Type annotate public APIs • ubuntu_desktop_installer/lib/tryorinstallpage.dart:37:7 • type_annotate_public_apis
   info • Type annotate public APIs • ubuntu_desktop_installer/lib/tryorinstallpage.dart:38:7 • type_annotate_public_apis
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/tryorinstallpage.dart:79:21 • 	
   info • Avoid lines longer than 80 characters • ubuntu_desktop_installer/lib/tryorinstallpage.dart:121:1 • lines_longer_than_80_chars
   info • Avoid lines longer than 80 characters • ubuntu_desktop_installer/lib/tryorinstallpage.dart:129:1 • lines_longer_than_80_chars
   info • Type annotate public APIs • ubuntu_desktop_installer/lib/tryorinstallpage.dart:131:28 • type_annotate_public_apis ????
   info • Type annotate public APIs • ubuntu_desktop_installer/lib/tryorinstallpage.dart:139:7 • type_annotate_public_apis
   info • Don't create a lambda when a tear-off will do • ubuntu_desktop_installer/lib/tryorinstallpage.dart:225:36 • unnecessary_lambdas
   info • Avoid lines longer than 80 characters • ubuntu_desktop_installer/lib/turnoffrstpage.dart:19:1 • lines_longer_than_80_chars
   info • Type annotate public APIs • ubuntu_desktop_installer/lib/turnoffrstpage.dart:21:23 • type_annotate_public_apis
   info • Don't create a lambda when a tear-off will do • ubuntu_desktop_installer/lib/turnoffrstpage.dart:54:30 • unnecessary_lambdas
   info • Prefer relative imports for files in `lib/` • ubuntu_desktop_installer/lib/welcomepage.dart:6:8 • prefer_relative_imports
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/welcomepage.dart:82:39 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/lib/welcomepage.dart:82:61 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • ubuntu_desktop_installer/test/widget_test.dart:16:8 • avoid_types_on_closure_parameters
```

</details>

File: ubuntu_desktop_installer/lib/i18n.dart [hunk 2]
```diff
@@ -17,14 +18,11 @@ class UbuntuLocalizations {
     });
   }
 
-  static UbuntuLocalizations of(BuildContext context) {
-    return Localizations.of<UbuntuLocalizations>(context, UbuntuLocalizations);
-  }
+  static UbuntuLocalizations of(BuildContext context) =>
```
Comment (MarcusTomlinson) [inline ubuntu_desktop_installer/lib/i18n.dart:21]: I'm not convinced these {} to => changes are necessarily good practice. I'd prefer we kept the brackets. Please could you revert all of these.

**Review by MarcusTomlinson — CHANGES_REQUESTED** [review]

File: ubuntu_desktop_installer/lib/i18n.dart (hunk 2 above)
Comment (iapicca) [reply ubuntu_desktop_installer/lib/i18n.dart:21]: sure thing, I will

**Comment by iapicca** [comment]
> > @iapicca, so it appears there is a set of linter rules for Effective Dart :)
> > If we are to make the switch from pedantic to this (which I actually think we should), we best go all in.
> > Would you be up to fixing the rest of these analyser errors in your branch? https://github.com/canonical/ubuntu-desktop-installer/pull/18/checks?check_run_id=1834232513
> > Or would you prefer I address them?
> 
> I've done these
> 
> lint for ubuntu_desktop_installer
> tomorrow I'll do the remaining ones

I have to keep these 

<details>
<summary>NON lowerCamelCase to keep as is</summary>


```
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:21:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:22:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:23:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:24:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:25:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:26:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:27:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:28:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:29:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:30:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:31:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:35:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:36:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:37:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:38:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:39:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:43:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:44:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:45:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:46:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:47:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:48:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • subiquity_client/lib/src/types.dart:49:3 • constant_identifier_names
   ```

</details>

otherwise it breaks the api

**Comment by Abhishek01039** [comment]
image

You haven't changed this code @iapicca.

**Comment by iapicca** [comment]
> You haven't changed this code 

I don't know what you mean by that

> then also CI gives an error here.

it's not an error, it's `effective_dart` complaining about formatting,
somehow it ignores the comment I committed [here](https://github.com/canonical/ubuntu-desktop-installer/pull/18/commits/6899f382ae788c108926fee40528263bde907e0c)

**Comment by Abhishek01039** [comment]
I am just saying that CI has failed because of that issue.

**Comment by MarcusTomlinson** [comment]
@iapicca thanks so much for this! I've reverted a few more things I thought were questionable such as implicit vs explicit variables, and a few more `{}` to` =>` changes. I also think it's a little saner to trust the dart formatter to reduce line lengths rather than make it an analysis issue. All looks good to me now. Merging...

**Pull request merged**